src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs

```diff
@@ -18,40 +17,58 @@ public static int[] GetProcessIds()
-            };
-            dummy._threadInfoList.Add(new ThreadInfo()
+                throw new ArgumentOutOfRangeException("pid");
+            }
+
+            ProcessInfo procInfo = new ProcessInfo();
+
+            // Try to get the task info. This can fail if the user permissions don't permit
+            // this user context to query the specified process
+            Interop.libproc.proc_taskallinfo? info = Interop.libproc.GetProcessInfoById(pid);
+            if (info.HasValue == false)
             {
-                _basePriority = 0,
-                _currentPriority = 0,
-                _processId = pid,
-                _startAddress = IntPtr.Zero,
-                _threadId = 0,
-                _threadState = 0,
-                _threadWaitReason = ThreadWaitReason.Unknown
-            });
-
-            // TODO: Implement this
-            throw NotImplemented.ByDesign;
+                procInfo.BasePriority = info.Value.ptinfo.pti_priority;
+                procInfo.HandleCount = Interop.libproc.GetFileDescriptorsForPid(pid).Count;
+                procInfo.ProcessId = pid;
+                procInfo.ProcessName = info.Value.pbsd.pbi_comm;
+                procInfo.VirtualBytes = (long)info.Value.ptinfo.pti_virtual_size;
+                procInfo.WorkingSet = (long)info.Value.ptinfo.pti_resident_size;
+
+                // These values don't have meaning or don't exist on OSX
+                procInfo.PageFileBytes = -1;
+                procInfo.PoolNonpagedBytes = -1;
+                procInfo.PoolPagedBytes = -1;
+                procInfo.PrivateBytes = -1;
+                procInfo.VirtualBytesPeak = -1;
+                procInfo.SessionId = -1;
+            }
+
+            // Create a threadinfo for each thread in the process
+            Interop.libproc.GetAllThreadsInProcess(pid).ForEach(t =>
+                {
+                    if (t.Value.HasValue)
+                    {
+                        procInfo._threadInfoList.Add(new ThreadInfo()
+                            {
+                                _basePriority = 0,
+                                _currentPriority = t.Value.Value.pth_curpri,
+                                _processId = pid,
+                                _startAddress = IntPtr.Zero, // We don't have this info
+                                _threadId = Convert.ToInt32(t.Key),
+                                _threadState = Interop.libproc.ConvertOsxThreadRunStateToThreadState((Interop.libproc.ThreadRunState)t.Value.Value.pth_run_state),
+                                _threadWaitReason = Interop.libproc.ConvertOsxThreadFlagsToWaitReason((Interop.libproc.ThreadFlags)t.Value.Value.pth_flags)
+                            });
+                    }
+                });
+            
+            return procInfo;
         }
 
         // ----------------------------------
-        // ---- Unix PAL layer ends here ----
+        // ---- OSX PAL layer ends here ----
```

Comment:
This should stay "Unix".  The idea (and we can always change this later if we want) is that some functionality is that some functionality in the .Unix.cs file depends on additional functions being exposed from the Unix-specific implementations, i.e. .Linux.cs and .OSX.cs.  Anything so required would be above this comment, and anything specific to the OSX implementation only used by functionality above the comment would be below the comment.